pkg/osm/ingress.go

```diff
@@ -0,0 +1,26 @@
+package osm
+
+// SupportedIngress maintains a mapping of OSM version to supported Ingress resource versions.
+var SupportedIngress = map[ControllerVersion][]IngressVersion{
+	"v0.5": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.5/pkg/ingress/client.go#L6
+		"extensions/v1beta1",
+	},
+	"v0.6": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.6/pkg/ingress/client.go#L6
+		"extensions/v1beta1",
+	},
+	"v0.7": {
+		// Source: https://github.com/openservicemesh/osm/blob/release-v0.7/pkg/ingress/client.go#L6
+		"extensions/v1beta1",
+	},
```

Comment:
I don't see a reason to maintain releases prior to v0.7. These are not even documented anymore and as such are not supported. Let's remove this.

Reply:
We do have release-v0.* branches. Perhaps we should clean these up too.
I'm happy to delete some of these items at some point if we find them unnecessary.

Reply:
Release branches should not be removed. But we don't need to keep supporting older releases in code, such as this one.